sklearn/preprocessing/data.py

```diff
@@ -1917,3 +1920,38 @@ def transform(self, X):
         """
         return _transform_selected(X, self._transform,
                                    self.categorical_features, copy=True)
+
+
+def _boxcox(X):
+    X, _ = stats.boxcox(X)
+    return X
+
+
+def boxcox_transform(X, copy=True):
+    """BoxCox transform to the input data
+
+    Apply boxcox transform on individual features with lambda
+    that maximizes the log-likelihood function for each feature
+
+    Parameters
+    ----------
+    X : array-like, shape [n_samples, n_features]
+        The data to be transformed. Should contain only positive data.
+
+    copy : boolean, optional, default is True
+        set to False to perform inplace transformation and avoid a
+        copy (if the input is already a numpy array or a scipy.sparse
+        CSR matrix and if axis is 1).
+
+    References
+    ----------
+    G.E.P. Box and D.R. Cox, "An Analysis of Transformations", Journal of the
+    Royal Statistical Society B, 26, 211-252 (1964).
+    """
+    X = check_array(X, ensure_2d=True, dtype=FLOAT_DTYPES)
+    if any(np.any(X<=0, axis=0)):
+        raise ValueError("BoxCox transform can only be applied on positive data")
+    n_samples, n_features = X.shape
+    outputs = Parallel()(delayed(_boxcox)(X[:, i]) for i in range(n_features))
+    output = np.asarray(outputs).T
```

Comment:
Sorry but I couldn't concatenate the 1D arrays on axis=1. So either I need to reshape the outputs to 2D and use concatenate or I could think was something like ``np.vstack(outputs).T``. Would it be efficient enough ?